### PR TITLE
feat(fixed_tvpi): implement partially binary operation with variable operands

### PR DESCRIPTION
**Pull request by LinerSu**
A rewrite rule for binary operation `x = y * z` if `y` or `z`'s value is constant.

### CODE REVIEW

File: include/crab/domains/fixed_tvpi_domain.hpp [hunk 1]
```diff
@@ -213,6 +213,26 @@ class fixed_tvpi_domain
...
+
+  void rewrite_apply_var(arith_operation_t op, const variable_t &x,
+                         const variable_t &y, const variable_t &z,
+                         unsigned coefficient) {
+    assert(coefficient > 1);
+
+    if (boost::optional<number_t> n = is_var_constant(z)) {
+      rewrite_apply(op, x, y, *n, coefficient);
+    } else if (boost::optional<number_t> n = is_var_constant(y)) {
+      rewrite_apply(op, x, z, *n, coefficient);
```
Comment (caballa) [inline include/crab/domains/fixed_tvpi_domain.hpp]: This assumes that `op` is commutative so it won't work for subtraction and division.

Comment (LinerSu) [reply include/crab/domains/fixed_tvpi_domain.hpp]: Yes, you are right. If `y` is some constant `n`, then we should have a special rewrite for `x := n - z` or `x := n / z`. How about adding constraints for those two cases?

Comment (caballa) [reply include/crab/domains/fixed_tvpi_domain.hpp]: Unless we need those cases now I would just exclude them (i.e., do nothing if sub or div)

Comment (LinerSu) [reply include/crab/domains/fixed_tvpi_domain.hpp]: Ok. Let me add one condition based on this.

Comment (LinerSu) [reply include/crab/domains/fixed_tvpi_domain.hpp]: DONE.

File: include/crab/domains/fixed_tvpi_domain.hpp [hunk 2]
```diff
@@ -213,6 +213,29 @@ class fixed_tvpi_domain
...
+  void rewrite_apply_var(arith_operation_t op, const variable_t &x,
+                         const variable_t &y, const variable_t &z,
+                         unsigned coefficient) {
+    assert(coefficient > 1);
+
+    if (boost::optional<number_t> n = is_var_constant(z)) {
+      rewrite_apply(op, x, y, *n, coefficient);
+    } else if (op == OP_ADDITION || OP_MULTIPLICATION) {
+      // Ignore case if op is subtraction and division
+      if (boost::optional<number_t> n = is_var_constant(y)) {
```
Comment (caballa) [inline include/crab/domains/fixed_tvpi_domain.hpp:230]: you forgot `m_base_absval -= ghost_x;` in case of sub or div

Comment (LinerSu) [reply include/crab/domains/fixed_tvpi_domain.hpp:230]: fixed

**Pull request merged**